contracts/contracts/GameItems.sol

```diff
@@ -1,32 +1,223 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
-import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
 import "./interfaces/IGameItems.sol";
 
-contract GameItems is ERC1155, AccessControl, IGameItems {
+/**
+*
+* GameItems for ETH CC
+*
+* @author Tobower
+* @author TheGreatAxios
+*
+* @dev This smart contract is an ERC-1155 standard contract
+* @dev This contract is the core tokens for the mini-game
+*
+**/
+contract GameItems is AccessControlEnumerable, ERC1155URIStorage, IGameItems {
+
+    /// @notice Admin Role to Toggle Contract
     bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    /// @notice Can Fire Initial Mint
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    /// @notice Can Mint on Wini
+    bytes32 public constant WIN_MANAGER_ROLE = keccak256("WIN_MANAGER_ROLE");
+
+    /// @notice Number of Items on the contract
+    uint256 private numberItems;
+    /// @notice Number of Active Players
+    uint256 private numberPlayers;
+    /// @notice The storage of the shared items on-chain #SKALE
+    mapping(uint256 => GameItemMetadata) private metadata;
+
+    /// @notice Modifier Error String
+    string private constant ACCESS_DENIED = "Access Denied";
```

Comment:
Solidity now has “error” for this kind of thing, but mostly for gas and we don’t care about that :). Just pointing it out.

Reply:
Correct, the reason I am confident it is ok is because some of the core SKALE contracts themselves use this pattern. 
Example: ConfigController. 

Showcases SKALE in another way by showing that you can do things like this. While it may not be gas friendly on Ethereum, since we don't care about it, it makes the contract a bit cleaner and organized IMO. Can always be changed to inline.